pkg/common/configs/config.go

```diff
@@ -194,15 +199,53 @@ func GetConfigurationString(requestBytes []byte) string {
 	return strings.ReplaceAll(conf, checksum, "")
 }
 
-// DefaultSchedulerConfig contains the default scheduler configuration; used if no other is provided
-var DefaultSchedulerConfig = `
-partitions:
-  - name: default
-    placementrules:
-      - name: tag
-        value: namespace
-        create: true
-    queues:
-      - name: root
-        submitacl: '*'
-`
+// GetDefaultConfig creates default scheduler configuration; used if no other is provided
+func GetDefaultConfig() (string, error) {
+	config := &SchedulerConfig{
+		Partitions:           GetDefaultPartitionConfig(),
+		EventRequestsEnabled: DefaultEventRequestsEnabled,
+		EventTrackingEnabled: DefaultEventTrackingEnabled,
+		EventRequestCapacity: DefaultEventRequestCapacity,
+		EventBufferCapacity:  DefaultEventBufferCapacity,
+		EventResponseSize:    DefaultEventResponseSize,
+	}
+
+	// ideally err will always be nil as we are now building the conf programatically
+	content, err := yaml.Marshal(config)
+	if err != nil {
+		log.Log(log.Config).Error("failed to marshal config object",
+			zap.Error(err))
+		return "", err
```

Comment:
We should definitely not return an empty config here. I can imagine two solutions: you either store the serialized YAML again (which you validate in the unit tests) or just simply call `panic()`. I vote for storing the YAML in the string literal, I think it's perfectly fine for a default config.
I just checked there is no test case for it, but you can add one which makes sure that it's a valid one.

Reply:
I think serialized YAML should be fine. This path should ideally not get exercised as we are now programmatically building the default config, we do something horribly bad with the default values. I'll make the changes and see if others have a different opinion.

Reply:
I initially agreed on storing the serialized YAML. But I think if we do that we have to re-introduce the `DefaultConfigString` which I got rid of as part of this change. If we bring that back, I think we should just have the `DefaultConfigString` and completely remove `GetDefaultConfig()`

When adding new configs, there are more chances of missing out on updating `DefaultConfigString` if we have to update default values in multiple places. We might end up in inconsistent default config in this case.

I'd like to go the previous behavior or calling `panic()` on error. Let's see what others think before jumping on the solution.

Reply:
I added a top-level comment, but SchedulerConfig and these event configurations have nothing to do with one another, and no changes should be made to SchedulerConfig. Instead, have the event system register for configmap changes (see the health check system for how to do that). Whenever the configmap changes, the callbacks get run, and each subsystem can then parse out and apply any configuration, falling back to default values for anything not set.